CONTRIBUTING.md

```diff
@@ -149,335 +231,329 @@ and dependencies.
+└── assets/                         # Optional: images, diagrams, etc.
+```
 
-You can then build that operator by configuring your build with `-D OP_my_operator:BOOL=1` (or `-D EXT_my_extension:BOOL=1` for GXF extensions):
+#### Packages
 
-```bash
-cmake -S . -B ./build -D OP_my_operator:BOOL=1
-cmake --build ./build -j
+```text
+holohub/pkg/your_package_name/
+├── CMakeLists.txt                  # Required: package configuration
+└── README.md                       # Optional: package description
 ```
 
-#### Operator Naming Convention
+### Naming Conventions
 
-For an operator named "Adaptive Thresholding", follow these naming conventions:
+For an operator named "Adaptive Thresholding":
 
-1. **Class Name**: `AdaptiveThresholdingOp` (title case without spaces and with "Op" at the end)
-2. **metadata.json ("name")**:  Same as class name.
-3. **Directory**: `adaptive_thresholding` (snake case)
-4. **Filename**: same as directory name with extension.
-5. **README (Title)**: "`Adaptive Thresholding Operator`" (title case with spaces and Operator at the end).
-6. **Unit testing**: `test_adaptive_thresholding.py` (for Python operators)
+| Component | Convention | Example |
+|-----------|------------|---------|
+| Class Name | TitleCase + "Op" suffix | `AdaptiveThresholdingOp` |
+| metadata.json "name" | Same as class name | `AdaptiveThresholdingOp` |
+| Directory | snake_case | `adaptive_thresholding` |
+| Filename | Same as directory + extension | `adaptive_thresholding.py` |
+| README Title | Title Case + "Operator" | "Adaptive Thresholding Operator" |
+| Unit Test | "test_" + directory name | `test_adaptive_thresholding.py` |
 
-### Adding an Application
+### Build System Integration
 
-Add each application in its own subdirectory under the [`applications`](./applications/) directory. The subdirectory should contain:
+All contributions that include code need to be integrated with HoloHub's build system using CMake. Edit the appropriate `CMakeLists.txt` to add your contribution:
 
-- A _metadata.json_ file which describes its specifications and requirements in accordance with the [application metadata.json schema](./applications/metadata.schema.json).
-- A README file summarizing the application's purpose and architecture;
-
-Edit [```CMakeLists.txt```](./applications/CMakeLists.txt) to add the new application as part of the build system using the ```add_holohub_application```
-CMake function, passing the new application folder name as the first argument. If the application relies on one or more operators then the optional ```DEPENDS OPERATORS``` should be added so that
-the build system knows to build the dependent operator(s).
+**For Operators:**
 
 ```cmake
-add_holohub_application(my_application DEPENDS
-                        OPERATORS my_operator1
-                                  my_operator2
-                        )
+# In ./operators/CMakeLists.txt
+add_holohub_operator(my_operator DEPENDS EXTENSIONS my_extension)
```

Comment:
I see that an explanation of `DEPENDS` is removed in these changes, can we add a note referring devs to https://github.com/nvidia-holoscan/holohub/blob/main/cmake/HoloHubConfigHelpers.cmake for for function arguments? And possibly improve function documentation that file as well?

Reply:
Thanks for your comment. I’ve brought back the explanation, but I’m not sure if mentioning the CMake helper functions is necessary. I think it might confuse the contributor instead of helping them.